roles/matrix-riot-web/defaults/main.yml

```diff
@@ -28,3 +28,6 @@ matrix_riot_web_embedded_pages_home_path: ~
 
 # Controls whether the self-check feature should validate SSL certificates.
 matrix_riot_web_self_check_validate_certificates: true
+
+# cross use of variable defined in matrix-synapse role
+matrix_synapse_enable_registration: false
```

Comment:
Defining variables from another "namespace" is discouraged.

A better way to do this would be to define a new `matrix_riot_web_..` variable (e.g. `matrix_riot_web_registration_enabled: false` in `matrix-riot-web/defaults/main.yml`). This is just some sane default provided by the `matrix-riot-web` role.

For ease of use, the playbook can then override this role variable by wiring it to `matrix_synapse_enable_registration`. It's possible to accomplish this by defining something like this in the **riot-web** section of the playbook's `group_vars/matrix-servers` file:

```yaml
matrix_riot_web_registration_enabled: "{{ matrix_synapse_enable_registration }}"
```

The benefits of doing it this way are that:

- Each role can be used independently of the other roles because none of them cross-reference variables (some people use the roles separately, not as part of this integrated playbook)

- thanks to the playbook's wiring (in the `group_vars/matrix-servers` file), changing `matrix_synapse_enable_registration` automatically reflects to `matrix_riot_web_registration_enabled` - a convenience feature for users of the playbook

- if necessary, `matrix_riot_web_registration_enabled` can also be controlled separately -- people can override it and set an explicit value (one that breaks the connection to `matrix_synapse_enable_registration`)

----------

You would probably notice the inconsistency in naming that I'm suggesting (`matrix_riot_web_registration_enabled`, instead of `matrix_riot_web_enable_registration`).

Generally, we aim to go for naming like `matrix_riot_web_registration_enabled`, so my proposal is consistent with that.
A lot of roles use variables named following this pattern.

The Synapse `matrix_synapse_enable_registration` variable is an outlier, because it's built like this:
- obligatory `matrix_synapse_` prefix, because it resides in the `matrix-synapse` role
- `enable_registration` suffix which is identical to the variable it controls in `homeserver.yaml`